packages/kbn-optimizer/src/optimizer/cache_keys.test.ts

```diff
@@ -0,0 +1,208 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import Path from 'path';
+
+import jestDiff from 'jest-diff';
+import { REPO_ROOT, createAbsolutePathSerializer } from '@kbn/dev-utils';
+
+import { reformatJestDiff, getOptimizerCacheKey, diffCacheKey } from './cache_keys';
+import { OptimizerConfig } from './optimizer_config';
+
+jest.mock('./get_changes.ts', () => ({
+  getChanges: async () =>
+    new Map([
+      ['/foo/bar/a', 'modified'],
+      ['/foo/bar/b', 'modified'],
+      ['/foo/bar/c', 'deleted'],
+    ]),
+}));
+
+jest.mock('./get_mtimes.ts', () => ({
+  getMtimes: async (paths: string[]) => new Map(paths.map(path => [path, 12345])),
+}));
+
+jest.mock('execa');
+
+jest.mock('fs', () => {
+  const realFs = jest.requireActual('fs');
+  return {
+    ...realFs,
+    readFile: jest.fn(realFs.readFile),
+  };
+});
```

Comment:
Unfortunately the jest tests aren't very well isolated so when trying to run them in the flaky test suite we get all sorts of errors about file system issues, http ports already bound, etc. Because of this I think we can just go off our hunch that `fs` isn't reset like the other modules between test files and see how it goes over time.

Reply:
Looks like an alternative would be to call `resetModules` as a heavy handed solution. Thoughts on calling that in `src/dev/jest/setup/mocks.js` as defined by `setupFilesAfterEnv`

https://github.com/facebook/jest/issues/3577#issuecomment-301627789

Reply:
I'm not confident in the ramifications of a change like that; I'm not sure what problems that would cause...